docs/nd4j/templates/quickstart.md

```diff
@@ -0,0 +1,381 @@
+
+// For higher dimensions you can provide a shape array. 2D random matrix example:
+int rows = 4;
+int cols = 5;
+int[] shape = {rows, cols};
+INDArray x = Nd4j.rand(shape);
+```
+
+Use the `arange` functions to create an array of evenly spaces values:
+
+```java
+INDArray  x = Nd4j.arange(5);
+// [         0,    1.0000,    2.0000,    3.0000,    4.0000]
+
+INDArray  x = Nd4j.arange(2, 7);
+// [    2.0000,    3.0000,    4.0000,    5.0000,    6.0000]
+```
+
+The `linspace` function allows you to specify the number of points generated:
+```java
+INDArray  x = Nd4j.linspace(1, 10, 5);
+// [    1.0000,    3.2500,    5.5000,    7.7500,   10.0000]
+
+// Evaluate a function over many points.
+import static org.nd4j.linalg.ops.transforms.Transforms.sin;
+INDArray  x = Nd4j.linspace(0.0, Math.PI, 100, DataType.DOUBLE);
+INDArray  y = sin(x);  
+```
+
+### Printing Arrays
+The INDArray supports Java's `toString()` method. The output is similar to printing NumPy arrays:
+```java
+INDArray  x = Nd4j.arange(6);  //1d array
+System.out.println(x);
+// [         0,    1.0000,    2.0000,    3.0000,    4.0000,    5.0000]
+
+int [] shape = {4,3};
+x = Nd4j.arange(12).reshape(shape);   //2d array
+System.out.println(x);
+/*
+[[         0,    1.0000,    2.0000], 
+ [    3.0000,    4.0000,    5.0000], 
+ [    6.0000,    7.0000,    8.0000], 
+ [    9.0000,   10.0000,   11.0000]]
+*/
+
+int [] shape2 = {2,3,4};
+x = Nd4j.arange(24).reshape(shape2);  //3d array
+System.out.println(x);
+/*
+[[[         0,    1.0000,    2.0000,    3.0000], 
+  [    4.0000,    5.0000,    6.0000,    7.0000], 
+  [    8.0000,    9.0000,   10.0000,   11.0000]], 
+
+ [[   12.0000,   13.0000,   14.0000,   15.0000], 
+  [   16.0000,   17.0000,   18.0000,   19.0000], 
+  [   20.0000,   21.0000,   22.0000,   23.0000]]]
+*/
+```
+
+### Basic Operations
+You will have to use INDArray methods to perform operations on your arrays. There are  in-place and copy overloads and scalar and element wise overloaded versions. The in-place operators return a reference to the array so you can conveniently chain operations together.
+
+```java
+//Copy
+arr_new = arr.add(scalar);    // return a new array with scalar added to each element of arr.
+arr_new = arr.add(other_arr); // return a new array with element wise addition of arr and other_arr.
+
+//in place.
+arr_new = arr.addi(scalar); //Heads up: arr_new points to the same array as arr.
+arr_new = arr.addi(other_arr);
+```
+
+addition: arr.add(...), arr.addi(...)
+substraction: arr.sub(...), arr.subi(...)
+multiplication: arr.mul(...), arr.muli(...)
+division: arr.div(...), arr.divi(...)
+
+When you perform the basic operations you must make sure the underlying data types are the same.
+```java
+int [] shape = {5};
+INDArray  x = Nd4j.zeros(shape, DataType.DOUBLE);
+INDArray  x2 = Nd4j.zeros(shape, DataType.INT);
+INDArray  x3 = x.add(x2);
+// java.lang.IllegalArgumentException: Op.X and Op.Y must have the same data type, but got INT vs DOUBLE
+```
```

Comment:
Show casting example as solution to this:
`INDArray x3 = x.add(x2.cast(DataType.DOUBLE))`

Reply:
`INDArray x3 = x.add(x2.castTo(DataType.DOUBLE));`